group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -579,4 +618,32 @@ public void shutdown() {
         Utils.closeQuietly(runtime, "coordinator runtime");
         log.info("Shutdown complete.");
     }
+
+    private static boolean isGroupIdNotEmpty(String groupId) {
+        return groupId != null && !groupId.isEmpty();
+    }
+
+    private static Errors toResponseError(Errors appendError) {
```

Comment:
It may be better to inline this code because the handling could be different depending on the request type. If I remember correctly, it is slightly different for offset commits for instance.

Reply:
confirmed that `storeGroup` and `storeOffsets` have different handling.

this will still be shared amongst join/sync/leave group, so i'll rename this to `appendGroupMetadataErrorToResponseError`, wdyt?

Reply:
That seems reasonable. We can see later if we could also share this logic with the consumer group heartbeat handling.